go/vt/vtgate/planbuilder/physical/route_planning.go

```diff
@@ -226,33 +226,47 @@ func createPhysicalOperatorFromDelete(ctx *plancontext.PlanningContext, op *abst
 		return nil, err
 	}
 
-	primaryVindex, vindexAndPredicates, err := getVindexInformation(op.TableID(), op.Table.Predicates, vindexTable)
-	if err != nil {
-		return nil, err
-	}
+	var r *Route
+	if opCode == engine.Unsharded {
+		r = &Route{
+			Source: &Delete{
+				QTable: op.Table,
+				VTable: vindexTable,
+				AST:    op.AST,
+			},
+			RouteOpCode:       opCode,
+			Keyspace:          vindexTable.Keyspace,
+			TargetDestination: dest,
+		}
```

Comment:
I think inside this `if` block, we should simply return the `*Route`. That way we can remove the entire `else`.

Reply:
Ah, yeah, good idea! 👍

Reply:
Done!